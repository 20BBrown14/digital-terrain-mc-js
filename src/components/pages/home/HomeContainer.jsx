import React from 'react';
import PropTypes from 'prop-types';
import HomeView from './HomeView';

const propTypes = {
  /* Function to navigate to new page */
  navigateToNewPage: PropTypes.func.isRequired,
};

/**
 * Home page container.
 */
const HomeContainer = (props) => {
  const { navigateToNewPage } = props;
  return (
    <HomeView
      navigateToNewPage={navigateToNewPage}
    />
  );
};

HomeContainer.propTypes = propTypes;

export default HomeContainer;
