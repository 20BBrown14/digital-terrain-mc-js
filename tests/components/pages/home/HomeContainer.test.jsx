import React from 'react';
import HomeContainer from '../../../../src/components/pages/home/HomeContainer';
import loadGalleryImagesService from '../../../../src/components/pages/gallery/service';

jest.mock('../../../../src/components/pages/gallery/service');

describe('HomeContainer', () => {
  afterEach(() => {
    loadGalleryImagesService.mockClear();
  });

  it('renders a default view', () => {
    const container = render(
      <HomeContainer
        navigateToNewPage={() => {}}
        homeCachedImages={[]}
        setCachedHomeImages={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('has a HomeView', () => {
    const container = shallow(
      <HomeContainer
        navigateToNewPage={() => {}}
        homeCachedImages={[]}
        setCachedHomeImages={() => {}}
      />,
    );
    expect(container.find('HomeView')).toHaveLength(1);
  });

  it('has a default state', () => {
    const container = shallow(
      <HomeContainer
        navigateToNewPage={() => {}}
        homeCachedImages={[]}
        setCachedHomeImages={() => {}}
      />,
    ).instance();

    const expectedState = {
      isCarouselLoading: true,
      isCarouselLoadingFailed: false,
      carouselImages: [],
    };

    expect(container.state).toEqual(expectedState);
  });

  describe('componentDidMount', () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <HomeContainer
          navigateToNewPage={() => {}}
          homeCachedImages={[]}
          setCachedHomeImages={() => {}}
        />,
      ).instance();
    });

    it('calls loadCarouselImages', () => {
      container.loadCarouselImages = jest.fn();
      container.componentDidMount();
      expect(container.loadCarouselImages).toHaveBeenCalledTimes(1);
    });
  });

  describe('loadCarouselImages', () => {
    describe('images are cached', () => {
      let container;
      beforeEach(() => {
        container = shallow(
          <HomeContainer
            navigateToNewPage={() => {}}
            homeCachedImages={[{}, {}, {}, {}]}
            setCachedHomeImages={() => {}}
          />,
        ).instance();
      });

      it('sets state as expected', () => {
        container.state.isCarouselLoadingFailed = true;
        container.loadCarouselImages();
        expect(container.state.isCarouselLoading).toEqual(false);
        expect(container.state.isCarouselLoadingFailed).toEqual(false);
        expect(container.state.carouselImages).toEqual([{}, {}, {}, {}]);
      });
    });

    describe('images are not cached', () => {
      let container;
      let mockSetCachedHomeImages;
      beforeEach(() => {
        mockSetCachedHomeImages = jest.fn();
        container = shallow(
          <HomeContainer
            navigateToNewPage={() => {}}
            homeCachedImages={[]}
            setCachedHomeImages={mockSetCachedHomeImages}
          />,
        ).instance();
      });

      it('calls loadGalleryImagesService', () => {
        loadGalleryImagesService.mockClear();
        container.loadCarouselImages();
        expect(loadGalleryImagesService).toHaveBeenCalledTimes(1);
        expect(loadGalleryImagesService).toHaveBeenCalledWith(
          expect.any(Function),
          expect.any(Function),
          true,
        );
      });

      describe('successful call', () => {
        beforeEach(() => {
          loadGalleryImagesService.mockImplementation((success) => {
            success([{}, {}, {}, {}]);
          });
        });

        it('sets state as expected and calls setCachedHomeImages', () => {
          container.loadCarouselImages();
          expect(container.state.isCarouselLoading).toEqual(false);
          expect(container.state.isCarouselLoadingFailed).toEqual(false);
          expect(container.state.carouselImages).toEqual([{}, {}, {}, {}]);
          expect(mockSetCachedHomeImages).toHaveBeenCalledTimes(1);
          expect(mockSetCachedHomeImages).toHaveBeenCalledWith([{}, {}, {}, {}]);
        });
      });

      describe('failed call', () => {
        beforeEach(() => {
          loadGalleryImagesService.mockImplementation((success, fail) => {
            fail();
          });
        });

        it('sets state as expected', () => {
          container.loadCarouselImages();
          expect(container.state.isCarouselLoading).toEqual(false);
          expect(container.state.isCarouselLoadingFailed).toEqual(true);
        });
      });
    });
  });
});
