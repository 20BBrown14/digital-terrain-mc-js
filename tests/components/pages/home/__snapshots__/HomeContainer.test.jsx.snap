// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer renders a default view 1`] = `
Array [
  <h1
    class="ant-typography home-page-title"
  >
    Welcome to Digital Terrain
  </h1>,
  <div
    class="ant-space ant-space-vertical"
  >
    <div
      class="ant-space-item"
      style="margin-bottom:8px"
    >
      <span
        class="ant-typography home-page-subtitle"
      >
        Digital Terrain is a SMP server inspired by Hermitcraft. The server is based on a great community, trust, and fun. We aspire to have a playerbase of all play styles that work together in harmony to build great things and have an awesome time. The server is owned by
        <button
          class="home-page-link-button"
          id="pandaLink"
          type="button"
        >
          Panda_Brah
        </button>
        and
        <button
          class="home-page-link-button"
          id="dmLink"
          type="button"
        >
          DMDF_
        </button>
        . Alongside them we have a team of mods who help make sure everything is running smoothly.
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span
        class="ant-typography home-page-subtitle"
      >
        Be sure to checkout our
        <button
          class="home-page-link-button"
          id="galleryLink"
          type="button"
        >
          gallery
        </button>
        and
        <button
          class="home-page-link-button"
          id="mapLink"
          type="button"
        >
          map
        </button>
        to get a feel for the server; hopefully you&aposll find it&aposs a good place for you! If you decide to, you can
        <button
          class="home-page-link-button"
          id="applyLink"
          type="button"
        >
          apply here
        </button>
        to be a part of our server. Hope to play with you soon!
      </span>
    </div>
  </div>,
  <div
    class="ant-carousel"
  >
    <div
      class="slick-slider home-page-carousel slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="width:1100%;left:-100%"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline:none;width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline:none;width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline:none;width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline:none;width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline:none;width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="6"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="7"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="8"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="9"
            style="width:9.090909090909092%"
            tabindex="-1"
          >
            <div>
              <img
                alt=""
                src="test-file-stub"
                style="width:100%;display:inline-block"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots slick-dots-bottom"
        style="display:block"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
        <li
          class=""
        >
          <button>
            5
          </button>
        </li>
      </ul>
    </div>
  </div>,
]
`;
